### DISCUSSION TITLE
Generic TLS Support

### DISCUSSION THREAD

**Discussion by bvernham** [earlephilhower/arduino-pico#815] [Q&A]
I see there is the TLS support for Wifi but is there generic support which includes ethernet and/or cellular?

Thanks

Bruce

**Comment by earlephilhower** [comment]
Adding support for ENC28J and WIZNET wired ethernet is in the queue here, but nothing related to 4G.  I suggest you talk to your board vendor, they may ave a solution already for that.

Basically, if the network interface hooks into LWIP, things will "just work" with the WiFiClientSecure.  If they do not use LWIP, but hack into some Arduino-specific higher level, then you probably can make them work but it is unlikely they'll work out of the box.

**Comment by khoih-prog** [comment]
Another way to use TLS for Ethernet, while lwip_XXX (lwip_W5x00, lwip_ENC,etc.) not ready yet, is to temporarily use [**EthernetWebServer_SSL** library](https://github.com/khoih-prog/EthernetWebServer_SSL).

For `NB-IoT/LTE-M/GRPS` modules, you can try [**NB_Generic** library](https://github.com/khoih-prog/NB_Generic). I actually dropped the work on this library quite some time, and just use as-is.
